azurerm/_modules/aks/main.tf

```diff
@@ -73,7 +73,7 @@ resource "azurerm_kubernetes_cluster" "current" {
     }
 
     oms_agent {
-      enabled                    = true
+      enabled                    = var.enable_log_analytics
       log_analytics_workspace_id = azurerm_log_analytics_workspace.current.id
```

Comment:
If `azurerm_log_analytics_workspace` is made conditional using count, it will need to be made conditional here too and referenced as `azurerm_log_analytics_workspace.current[0]` accordingly.

Reply:
The same applies for the reference to it in `azurerm_log_analytics_solution`